src/please.go

```diff
@@ -676,6 +679,35 @@ func please(tid int, state *core.BuildState, parsePackageOnly bool, include, exc
 	}
 }
 
+// set the watch
+func setWatchedTarget(state *core.BuildState, labels []core.BuildLabel) string {
+	if opts.Watch.Run {
+		if len(labels) > 1 {
+			opts.Run.Parallel.PositionalArgs.Targets = opts.Watch.Args.Targets
+		} else {
+			opts.Run.Args.Target = opts.Watch.Args.Targets[0]
+		}
+		return "run"
+	}
+
+	for i, label := range labels {
+		if state.Graph.TargetOrDie(label).IsTest {
+			if i == 0 {
+				opts.Test.Args.Target = label
+			}
+			opts.Test.Args.Args = append(opts.Test.Args.Args, label.String())
```

Comment:
you need to empty this slice somewhere, or it will grow unboundedly the more times the watch kicks in.

Reply:
not done?

It may be cleaner to do something like
```
if allTargetsAreTests(labels) {
    opts.Test.Args.Target = labels[0]
    opts.Test.Args.Args = labels[1:]
    return "test"
}
```

Reply:
Oh, I have moved the call to this function to the `watch` function, so it only gets called once and no longer needs clean up.